src/core/AudioEngine.cpp

```diff
@@ -337,6 +337,7 @@ void AudioEngine::pushInputFrames( sampleFrame * _ab, const f_cnt_t _frames )
 const surroundSampleFrame * AudioEngine::renderNextBuffer()
 {
 	m_profiler.startPeriod();
+	m_profiler.startDetail(AudioEngineProfiler::DetailType::NoteSetup);
```

Comment:
I read what RAII pattern is and wikipedia describes it in the context of classes, not functions. I suppose you meant to simply call the `startDetail` / `finishDetail` functions at the start and end of the separated stage functions?

It could make `AudioEngine::renderNextBuffer()` easier to navigate, although I'm not sure if it's even worth splitting off stage 2 (which is just 2 + 2 lines); that would have the opposite effect and just make the code a bit harder to find. :)

Reply:
I made new functions for the four stages, it probably does look a bit better this way. If that's not what you meant and / or this is an unnecessary change, the last commit can be simply reverted.

Reply:
I see, thanks for the explanation. I know about the concept from smart pointers and lock guards, but I never realized that this is what RAII was referring to. I'll think about it and try to apply it here, since half of the work (splitting stages into separate functions) is already done.

Reply:
OK, so after a few attempts it builds and actually works, so I hope my implementation is adequate. :)

I though about also using the RAII probes for the main timer (started and stopped by `startPeriod()` and `finishPeriod()`), but since it is using a different averaging weights and it can directly influence playback[^1], I would rather not touch it and keep it as a separate special case.

[^1]: By determining the cutoff, when LMMS considers performance insufficient and stops playing new notes. I personally don't think the current cutoff detection works particularly well, but that's for another issue / PR.